MAINT, DOC: bump jupyterlite-sphinx lower bound

tylerjereddy opened the pull request:
~~* Two versions of `jupyterlite-sphinx` were released in the last 5 hours, and neither one works with our doc build per gh-21212, so just pin for now. (I did confirm locally)~~

Update: just bump the lower bound for `jupyterlite-sphinx` now that there's an upstream fix.

[skip actions] [skip cirrus]

tylerjereddy committed:
CI: pin jupyterlite-sphinx

* Two versions of `jupyterlite-sphinx` were released in the last
5 hours, and neither one works with our doc build per gh-21212,
so just pin for now.

[skip actions] [skip cirrus]

melissawm commented on .circleci/config.yml:
```diff
@@ -116,6 +116,9 @@ jobs:
             export PYTHONPATH=$PWD/build-install/lib/python3.11/site-packages
+            # TODO: remove jupyterlite-sphinx pin when
+            # gh-21212 is resolved upstream
+            python -m pip install jupyterlite-sphinx==0.15.0
```
```suggestion
            python -m pip install jupyterlite-sphinx>=0.16.2
```

This should work now that 0.16.2 including the fix is in PyPI. Thanks!

tylerjereddy replied:
I guess since `pyproject.toml` (and `requirements/doc.txt` used by the CI here) contain `jupyterlite-sphinx>=0.13.1`, we could fix CI by just closing/ignoring this PR.

To be more correct we might want update `pyproject.toml` with your new version requirements and then use the dep propagation script to update the `requirements` files.

melissawm replied:
OK that may be safer just to be sure we don't install the wrong versions there. @steppi does that sound good?

tylerjereddy replied:
I went ahead and did that, I can't think of anything better.

tylerjereddy committed:
MAINT: PR 21213 revisions

* unpin `jupyterlite-sphinx` now that there's an upstream fix;
instead, updated our requirements files to reflect a newer
minimum required version of `jupyterlite-sphinx`

[docs only]

melissawm commented:
Looks like this did the trick, thanks @tylerjereddy !

melissawm commented:
(Github is having an incident and the artifacts are not available so I'll approve and come back)

steppi commented:
conda-forge release of 0.16.2 is on hold due to azure outage holding up CI https://github.com/conda-forge/jupyterlite-sphinx-feedstock/pull/30#pullrequestreview-2187078933

steppi commented:
> conda-forge release of 0.16.2 is on hold due to azure outage holding up CI [conda-forge/jupyterlite-sphinx-feedstock#30 (review)](https://github.com/conda-forge/jupyterlite-sphinx-feedstock/pull/30#pullrequestreview-2187078933)

Outage is over and conda-forge release is done https://github.com/conda-forge/jupyterlite-sphinx-feedstock/pull/30#event-13566271590

melissawm commented:
All looks good, thanks folks!
